src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,121 @@
+package duke;
+
+/**
+ * Parser executes all actions passed from the Ui class
+ */
+public class Parser {
+
+    TaskList tasks;
+
+    Parser(TaskList tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * prints the list of currently stored tasks
+     */
+    String listTasks() {
+        String response = "Here are the tasks in your list:\n";
+        for (int i = 1; i <= tasks.size(); i++) {
+            response += i + ". " + tasks.get(i - 1) + "\n";
+        }
+        return response;
+    }
+
+
+    /**
+     * finds and prints task based on whether it contains the given keyword
+     *
+     * @param keyword word to be matched with task description
+     */
+    String findTask(String keyword) {
+        String response = "Here are the matching tasks in your list:\n";
+        int currCount = 1;
+        for (int i = 1; i <= tasks.size(); i++) {
+            if (tasks.get(i - 1).getDescription().contains(keyword)) {
+                response += currCount++ + ". " + tasks.get(i - 1) +"\n";
+            }
+
+        }
+        return response;
+    }
+
+
+    /**
+     * marks the chosen task as done and prints a confirmation
+     *
+     * @param taskNumber represents which task to mark, 1 for the first task, 2 for the second, etc
+     */
```

Comment:
Adding the `@return` tag to public javadoc comments could help readers better understand the method, and, as an added bonus, more useful information will be shared by IDEs in their quick documentation feature.